lib/db/lowlevel.go

```diff
@@ -499,6 +492,30 @@ func (db *Lowlevel) checkGlobals(folder []byte, meta *metadataTracker) error {
 	return t.Commit()
 }
 
+func checkGlobalsFilterDevices(dk, folder, name []byte, devices [][]byte, vl VersionList, t readOnlyTransaction) (VersionList, bool, error) {
+	var changed bool
+	var err error
+	for _, device := range devices {
+		dk, err = t.keyer.GenerateDeviceFileKey(dk, folder, device, name)
+		if err != nil {
+			return vl, false, err
+		}
+		f, ok, err := t.getFileTrunc(dk, true)
+		if err != nil {
+			return vl, false, err
+		}
+		if !ok {
+			changed = changed || ok
+			continue
+		}
+		vl, _, _, _, _, _, _, err = vl.update(folder, device, f, t)
```

Comment:
this is literally a caterpillar emoji 🐛 

that's probably a red flag of some kind :D

Reply:
A green caterpillar though, so it's fine?! :P 

We need all the return values in `updateGlobal`. We could call `getGlobal` before and after there to shorten the caterpillar by 4 elements, but that would be redundant (as it's exactly what `update` does internally).